docs/data/date-pickers/custom-components/custom-components.md

```diff
@@ -40,50 +40,62 @@ To modify components position, have a look at the [custom layout](/x/react-date-
 ### Recommended usage
 
 :::success
-Remember to pass a reference to the component instead of an inline render function and define it outside of the main component.
-This ensures that the component is not re-rendered on every update.
+Remember to pass a reference to the component instead of an inline render function and define it outside the main component.
+This ensures that the component is not remounted on every update.
 :::
 
-This code is buggy because it will render a new input after each keystroke, leading to a loss of focus.
+Both slots in the example below are buggy because they will remount after each keystroke, leading to a loss of focus.
 
 ```jsx
+// ❌ The `toolbar` slot is re-defined each time the parent component renders,
```

Comment:
I moved to 3 examples, always using the same slot
It took me some time reading the example to understand why we had 2 slots, I think the new structures is clearer.

Please let me know if you disagree

Reply:
I do agree, three concise examples make it more clear and easier to grok. 👍